traffic_ops/traffic_ops_golang/crconfig/deliveryservice.go

```diff
@@ -145,7 +145,7 @@ FROM deliveryservice AS d
 INNER JOIN type AS t ON t.id = d.type
 LEFT OUTER JOIN profile AS p ON p.id = d.profile
 WHERE d.cdn_id = (select id FROM cdn WHERE name = $1)
-AND d.active = true
+AND d.active = 'ACTIVE'
```

Comment:
The point of using a constant is to avoid misspellings and so you can change it once to change it everywhere - but this being an enum type in the database as well as a Go constant sort of escapes both of those use cases. Because it's an enum it's not possible to insert a misspelling; that would break the API and fail tests etc. And also if the string used to represent "Active" ever changes, it's not possible to change it in only one place, since it would require a database migration. Plus, using string concatenation for building SQL queries - besides always making me a little nervous inherently - means that knowing what you're doing is safe depends on knowing the exact string value of the constant, so an abstraction could be detrimental to that.

For all of those reasons I chose deliberately not to use the Go constant in queries. If you still want me to change it I will, I just wanted to explain my reasoning.

Reply:
> And also if the string used to represent "Active" ever changes, it's not possible to change it in only one place, since it would require a database migration

Changing all of the Go occurrences in one place would still be possible

> Plus, using string concatenation for building SQL queries - besides always making me a little nervous inherently - means that knowing what you're doing is safe depends on knowing the exact string value of the constant, so an abstraction could be detrimental to that.

3 of the 6 queries I commented on are queries that already use concatenation, so concatenation should work there, as well as the 2 query fragments.

For the 3 queries that don't use concatenation, passing `tc.DSActiveStateActive` as a parameter to the prepared statement would work (and, unlike the other cases, would not require a typecast).

That said, if you feel that it's more maintainable as-is, I'm fine with you keeping it like that.

Reply:
> Changing all of the Go occurrences in one place would still be possible

Sure, but doing that yields a fundamentally broken piece of software, so idk how useful that is.

> 3 of the 6 queries I commented on are queries that already use concatenation, so concatenation should work there, as well as the 2 query fragments.

Without looking at those concatenations, I would say I probably disagree with their use as well, since I know such instances exist.

> For the 3 queries that don't use concatenation, passing tc.DSActiveStateActive as a parameter to the prepared statement would work (and, unlike the other cases, would not require a typecast).
>
> That said, if you feel that it's more maintainable as-is, I'm fine with you keeping it like that.

I think that's actually probably the best course, is to pass it in for statement preparation. That seems more proper than string concatenation and makes the connection between the Go constant and the database enum even more clear. So that's what I'll do.